### PR TITLE
executor: show memory consumption in slow query log

### PR DESCRIPTION
**Pull request by qw4990**
### What problem does this PR solve? 
Add memory consumption in slow query log.

### What is changed and how it works?
1. Add `maxBytes` into `mem.Tracker` to trace max number of bytes memory consumed during execution.
2. Update `SlowLogFormat` to add memory consumption in slow log.

### Check List 

Tests 

 - Unit test

### CODE REVIEW

File: sessionctx/variable/session.go [hunk 1]
```diff
@@ -897,6 +897,8 @@ const (
 	SlowLogQuerySQLStr = "Query" // use for slow log table, slow log will not print this field name but print sql directly.
 	// SlowLogStatsInfoStr is plan stats info.
 	SlowLogStatsInfoStr = "Stats"
+	// SlowLogMemUse is the number bytes of memory used in this statement.
+	SlowLogMemUse = "Mem_use"
```
Comment (zz-jason) [inline sessionctx/variable/session.go]: how about s/Mem_use/Mem_max/?

Comment (qw4990) [reply sessionctx/variable/session.go]: It has been updated.

**Comment by qw4990** [comment]
All conflicts have been resolved, PTAL.

**Pull request merged**